File: go/atcoder/ABC147/d.go
package main

import (
	"fmt"
	"math"
)

func main() {
	var n int
	fmt.Scan(&n)
	an := make([]int64, n)
	for i := 0; i < n; i++ {
		fmt.Scan(&an[i])
	}

	mod := int64(math.Pow(10, 9) + 7)
	var ans int64
	for d := 0; d <= 60; d++ {
		var count0 int64
		var count1 int64
		for _, a := range an {
			if (a>>uint(d))&1 == 1 {
				count1++
			} else {
				count0++
			}
		}

		keta := (1 << uint(d)) % mod
		count := (count0 * count1) * keta % mod
		ans += count
		ans %= mod
	}

	fmt.Println(ans)
}

File: ruby/atcoder/ABC147/d.rb
n = gets.to_i
an = gets.strip.split.map(&:to_i)

mod = 10 ** 9 + 7
ans = 0
keta = 1
(0..60).each do |d|
  count = an.count {|a| (a / keta) % 2 == 1}
  ans += count * (n - count) * keta
  ans %= mod
  keta *= 2
end

puts ans
